src/app/services/student-round.service.ts

```diff
@@ -145,6 +145,45 @@ export class StudentRoundService {
     shareReplay(1),
   );
 
+  interactions$: Observable<Interaction[]> =
+  combineLatest([this.sessionService.currentRoundPath$, this.authService.currentUser$]).pipe(
+    switchMap(([path, user]) =>
+      path && user
+        ? this.firebaseService.getStudentInteractions(path, user.uid)
+        : of([])),
+    distinctUntilChanged()
```

Comment:
Is `distinctUntilChanged()` actually doing anything here for an array without specific checks for equality?

Reply:
Oh, good catch. Looks like it's just checking reference identity at the moment.